### PR TITLE
open tweet from opentable

### PR DESCRIPTION
**Pull request by kyububba**
Hi Olivier,

Let me know if you have any questions.

Thanks,

Kai

### CODE REVIEW

File: OpenTweet/TweetTableViewCell.swift [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        }
+    }
+    
+    func updateContentLabel() {
+        if let contentString = tweet?.content {
+            //search for hyperlinks using a data detector.  to make this clickable we'll need to override UILabel or use an UITextView, which is a little heavy
+            let linkDetector = try! NSDataDetector(types: NSTextCheckingResult.CheckingType.link.rawValue )
+            let attrString : NSMutableAttributedString = NSMutableAttributedString(string: contentString)
+
+            let urlMatches = linkDetector.matches(in: contentString, options: [], range: NSMakeRange(0, contentString.characters.count))
```
Comment (olarivain) [inline OpenTweet/TweetTableViewCell.swift]: do you think it's appropriate to run data detection over and over again on the main thread?
How would you optimize this if you had to?

File: OpenTweet/UIImageExtensions.swift [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    }
+    
+}
+
+var urlHandle: Int = 0
+
+public extension UIImageView {
+    
+    func setImage(fromURL url: String?, placeholder: UIImage) {
+        objc_setAssociatedObject(self, &urlHandle, url, objc_AssociationPolicy.OBJC_ASSOCIATION_RETAIN_NONATOMIC)
```
Comment (kyububba) [inline OpenTweet/UIImageExtensions.swift]: You can subclass it obviously.  Just a matter of keeping the URL around.

File: OpenTweet/TweetTableViewCell.swift (hunk 1 above)
Comment (olarivain) [inline OpenTweet/TweetTableViewCell.swift]: fair enough. how do you solve cell reuse and avoid races in this case?

Comment (kyububba) [inline OpenTweet/TweetTableViewCell.swift]: it's similar to the image loading case, you just need a way to compare state.  You have a cell class, just add a field there, something simple like a incrementing counter, or something unique to the tweet state, pass it through to the GCD closure and compare it to the state on the cell when the parsing is done.

**Pull request closed without merging**